consensus/state.go

```diff
@@ -1033,31 +1074,69 @@ func (cs *ConsensusState) AddProposalBlockPart(height int, part *types.Part) (ad
 		log.Info("Received complete proposal", "hash", cs.ProposalBlock.Hash())
 		if cs.Step == RoundStepPropose && cs.isProposalComplete() {
 			// Move onto the next step
-			go cs.EnterPrevote(height, cs.Round)
+			go cs.EnterPrevote(height, cs.Round, false)
 		} else if cs.Step == RoundStepCommit {
 			// If we're waiting on the proposal block...
-			cs.tryFinalizeCommit(height)
+			cs.tryFinalizeCommit(height, cs.Round)
```

Comment:
cs.Round is not necessarily the commit round.
I think we should make a new field CommitRound, set it upon EnterCommit(height,commitRound), and make tryFinalizeCommit/FinalizeCommit not take a round param.  Alternatively we can maybe assign cs.Round = commitRound in EnterCommit.